### PR TITLE
fix(platform): declare the injected _cluster values in system charts

### LINKED ISSUE #4030: kubevirt: helm lint fails on the chart because _cluster is missing from its values
### What happens

`helm lint packages/system/kubevirt` fails on `main`:

```text
[ERROR] templates/: template: cozy-kubevirt/templates/vm-exportproxy-tlsroute.yaml:1:13: executing "cozy-kubevirt/templates/vm-exportproxy-tlsroute.yaml" at <index .Values._cluster "root-host">: error calling index: index of untyped nil
```

`templates/vm-exportproxy-tlsroute.yaml` and `templates/vm-exportproxy-ingress.yaml` both start with `index .Values._cluster "root-host"`, and the chart's own `values.yaml` has no `_cluster` key. The platform always injects `_cluster` at install time, so the deployed chart is fine; a standalone `helm lint` or `helm template` without `--set _cluster.root-host=…` is not. Nothing in CI runs lint on this chart, so nothing is red; it only bites whoever renders the chart by hand or adds a lint step later.

### Fix shape

Either declare `_cluster: {}` in `values.yaml` and read the keys through `dig`/`default` so the two templates render nothing when the platform context is absent, or guard both templates with `{{- with .Values._cluster }}`. Same shape as the nil-safe walk the platform bundle already uses for its own optional blocks.

Seen while running the chart's unit tests for #4015; present at that PR's merge base and unrelated to it.

### PR DESCRIPTION
**Pull request by lexfrei**
## What this PR does

Ten charts under `packages/system` read `_cluster`, and `bucket` and `monitoring` also read `_namespace`. The platform injects both through the `cozystack-values` Secret, so installs never notice. Rendering a chart on its own does: `helm template packages/system/kubevirt` dies on `index of untyped nil` before it emits anything, and the `_namespace` readers die on `nil pointer evaluating interface {}.host`.

Each of the ten now declares the keys it reads, with an empty default. Injected values merge over chart defaults, so an installed release renders exactly what it rendered before. Checked per chart against a realistic injected `_cluster`/`_namespace`: six came out byte-identical, and the other four differ only in `randAlphaNum` secrets, which also differ between two renders of the same tree.

The declaration on its own is not worth much, so every chart also gets a suite that renders with nothing injected. The existing suites all set `_cluster` themselves and cannot see whether `values.yaml` still declares it. Drop a declaration and exactly that chart's suite goes red, with the error its comment quotes; I ran that for all twelve. None of the new suites carries a suite-level `templates:` list, on purpose: that list narrows what helm-unittest renders, and would hide the same breakage in a template added later.

`kubevirt-cdi` had no `tests/` and no `test:` target, so it gets both. `etcd-operator` reads `_cluster` too and stays as it is, since its one call site already guards with `| default dict`. `cozystack-basics` already declared the key.

Closes #4030.

### Not in this PR

No CI job runs `helm lint` or a standalone `helm template`, which is why none of this was red. A gate is worth proposing separately. Its value is narrow though: `_cluster` is always populated on a real install, and eight of these ten already had a suite rendering the affected template with `_cluster` supplied. The two with nothing were `kubevirt`, whose `vm-exportproxy-*` templates no test touched, and `kubevirt-cdi`, which had no tests at all.

`helm lint` still exits 1 on `keycloak` and `monitoring` with `chart metadata is missing these dependencies: cozy-lib`. Both carry a `charts/cozy-lib` symlink that `Chart.yaml` never declares. That one is not a two-chart item: 82 charts across `packages/{system,apps,extra,core}` fail lint the same way, which is invisible because no job runs `helm lint`. Separate defect, untouched here.

Fifteen charts outside `packages/system` break the same way: `apps/{bucket,foundationdb,harbor,kubernetes,mongodb,nats,opensearch,postgres,vpn}` and `extra/{bootbox,etcd,gateway,info,ingress,seaweedfs}`. `cozy-lib`'s own helpers already guard every access with `| default dict`, so routing those reads through the library would close the class instead of patching fifteen more values files.

`kubevirt`'s suite only pins that the routes stay absent with nothing injected. Nothing pins that they show up once `_cluster` selects the service.

### Screenshots

No UI change: the diff is ten `values.yaml` files, ten helm-unittest suites, and one package `Makefile`.

### Downstream repositories

Walked the trigger map in `docs/agents/contributing.md` file by file against this diff. No package is added, renamed or removed under `packages/apps` or `packages/extra`; `packages/core/platform/values.yaml` is untouched; no variant, bundle, component, release asset, namespace or `ApplicationDefinition` semantic changes; no `values.schema.json` and no version enum moves, so the provider's hand-written schemas and defaults are unaffected; nothing under `hack/` moves and no existing make target changes behaviour. The one `Makefile` edit adds a `test:` target to `packages/system/kubevirt-cdi`, matching the target the other nine already have, so `make generate` and the ccp skills that drive it see nothing new.

- [x] No downstream repository is affected by this change
- [ ] [cozystack/website](https://github.com/cozystack/website) - follow-up:
- [ ] [cozystack/terraform-provider-cozystack](https://github.com/cozystack/terraform-provider-cozystack) - follow-up:
- [ ] [cozystack/ansible-cozystack](https://github.com/cozystack/ansible-cozystack) - follow-up:
- [ ] [cozystack/ccp](https://github.com/cozystack/ccp) - follow-up:
- [ ] [cozystack/talm](https://github.com/cozystack/talm) - follow-up:
- [ ] [cozystack/cozyhr](https://github.com/cozystack/cozyhr) - follow-up:
- [ ] [cozystack/cozy-proxy](https://github.com/cozystack/cozy-proxy) - follow-up:
- [ ] [cozystack/cozystack-telemetry-server](https://github.com/cozystack/cozystack-telemetry-server) - follow-up:
- [ ] [cozystack/external-apps-example](https://github.com/cozystack/external-apps-example) - follow-up:
- [ ] [cozystack/examples](https://github.com/cozystack/examples) - follow-up:
- [ ] [cozystack/community](https://github.com/cozystack/community) - follow-up:

### Release note

```release-note
fix(platform): declare the platform-injected `_cluster` and `_namespace` values in the ten system charts that read them, so each chart renders with `helm template` outside a cluster instead of failing on a nil map. No change to what an installed release renders.
```

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved standalone Helm chart rendering when platform-provided values are unavailable.
  - Prevented rendering failures across system charts, including bucket, dashboard, Keycloak, monitoring, KubeVirt, and related components.

- **Tests**
  - Added comprehensive standalone rendering coverage for system charts.
  - Added validation for expected ingress, gateway, TLS route, and resource output under different configuration scenarios.
  - Added a chart test command for KubeVirt CDI.

### CODE REVIEW

File: packages/system/monitoring/values.yaml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+# Injected by the platform from the cozystack-values Secret. The empty
+# defaults keep the chart renderable on its own, outside a cluster.
+_cluster: {}
+_namespace: {}
```
Comment (IvanHunters) [inline packages/system/monitoring/values.yaml:4]: **[MINOR]** the empty-map default renders `%!s(<nil>)` where a host belongs

A bare `{}` makes `.Values._namespace.host` an untyped nil, and the call sites that build hostnames with `printf "%s"` write Go's own formatting error into the manifest rather than an empty string. Nine occurrences across three of the ten charts:

```
$ helm template monitoring packages/system/monitoring --namespace cozy-monitoring | grep -n '%!'
197:        - "alerta.%!s(<nil>)"
200:    - host: "alerta.%!s(<nil>)"
280:      root_url: "https://grafana.%!s(<nil>)"
337:        - host: "grafana.%!s(<nil>)"
349:        - "grafana.%!s(<nil>)"
$ helm template bucket packages/system/bucket --namespace cozy-bucket | grep -n '%!'
36:          value: "s3.%!s(<nil>)"
$ helm template keycloak packages/system/keycloak --namespace cozy-keycloak | grep -n '%!'
133:              value: https://keycloak.%!s(<nil>)
181:      - keycloak.%!s(<nil>)
184:  - host: keycloak.%!s(<nil>)
```

Inside `bucket` the two code paths disagree about what unset looks like: the Ingress rule comes out `host: cozystack.` because `{{ $host }}` interpolates nil as empty, while the Deployment env comes out `s3.%!s(<nil>)` because that one goes through `printf`. Nothing in the PR notices either: the new suites assert document counts only, and `kubeconform -strict -ignore-missing-schemas` reports every one of these renders valid, since a formatting error is still a legal string. So the follow-up `helm template` gate the PR proposes would go green on this too.

Seeding the one key each printf path reads fixes it. I ran `_cluster: {root-host: ""}` and `_namespace: {host: ""}` on the three charts in isolated checkouts: the artifact count drops to zero, every suite stays green (`keycloak` 6/6, `monitoring` 8/8, `bucket` 3/3), and the render under a realistic injected `_cluster`/`_namespace` stays byte-identical to this PR's, because an injected key still wins the merge.

File: packages/system/kubevirt/tests/standalone_render_test.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+# case that sets it itself renders fine whether or not values.yaml declares
+# the key. This case supplies nothing. Drop the declaration from values.yaml
+# and the render dies on "index of untyped nil" before a single assert runs;
+# the declaration is the only thing keeping the chart renderable outside a
+# cluster. No suite-level `templates:` list here on purpose: it would narrow
+# the render to the listed files and let a later template reintroduce the
+# same breakage unseen.
+
+tests:
+  - it: exposes no vm-exportproxy route when _cluster is absent
```
Comment (IvanHunters) [inline packages/system/kubevirt/tests/standalone_render_test.yaml:16]: **[MINOR]** the two asserts on `vm-exportproxy` cannot fail if those templates go permanently dead

`kubevirt-cdi` got a positive counterpart in this PR and `kubevirt` did not, and `kubevirt` is the one chart where nothing else in the tree renders the templates in question (`grep -rl vm-exportproxy packages/system/kubevirt/tests/` matches only the new suite). I forced both exposure gates to `false` so the templates emit nothing under any input, then ran each chart's whole suite:

```
$ review-helper mutate /tmp/pr-review-cozystack-cozystack-4102 --mutations gatemut.json
GAPS: 1   answered: 2 of 2
kubevirt: vm-exportproxy ingress can never render GAP: the suite stayed green without the fix
| Test Suites: 2 passed, 2 total
kubevirt-cdi: uploadproxy ingress can never render covered: the suite went red
| Test Suites: 1 failed, 0 passed, 1 total
```

The `count: 0` asserts are not entirely inert: I renamed a template out from under one and helm-unittest failed with `template "..." not exists or not selected in test suite`, so a delete or rename is caught. What they cannot separate is "correctly absent because `_cluster` is empty" from "never renders again under any input". The fourteen lines that close it are already written next door in `kubevirt-cdi/tests/standalone_render_test.yaml`.

**Review by IvanHunters — APPROVED** [review]
## Verdict

LGTM with non-blocking notes

Reviewed at `eef5fe690` against merge-base `86d62bbb2`.

The mechanism holds up. I could not find a way for this to change what an installed release renders: across five configuration corners times ten charts, base and head produce identical manifests apart from `randAlphaNum` secrets, and those differ between two renders of the same tree anyway. Both notes below are about the standalone render this PR creates, not about anything that reaches a cluster.

## Findings

- [MINOR] `packages/system/monitoring/values.yaml:4`, the empty-map default renders `%!s(<nil>)` where a host belongs
- [MINOR] `packages/system/kubevirt/tests/standalone_render_test.yaml:16`, the two asserts on `vm-exportproxy` cannot fail if those templates go permanently dead

## Claim mismatches

[PARTIAL] "82 charts across `packages/{system,apps,extra,core}` fail lint the same way". I count 33, and the tier breakdown differs. `find packages -maxdepth 4 -path "*/charts/cozy-lib"` returns 35 charts, 33 of which never declare the dependency in `Chart.yaml`, and `helm lint` puts those at 22 in `apps`, 8 in `extra`, 2 in `system`, 1 in `tests`, none in `core`. 85 charts fail `helm lint` for one reason or another on helm v4.0.4, which is probably where 82 came from. Does not change the decision to defer it: 33 is still not a two-chart item.

## Caveats

- I did not replay an upgrade through helm-controller, and a green render does not prove one converges. The argument standing in for it: since no rendered field changes, there is no newly set or newly conditional field for server-side apply to conflict on, no immutable field to diverge, no admission-defaulted field to drift.
- The new `monitoring` suite sets the release namespace to `cozy-monitoring`, while the chart's existing suites use `tenant-root`, which is where its HelmRelease actually gets created. No assert in the new suite depends on the namespace, so nothing is wrong today. A content assert added there later would be exercising a namespace the chart does not install into.
- Versions: helm v4.0.4, helm-unittest v1.0.3, kubeconform v0.7.0. All local against the clone, no cluster contacted.

## Recommended follow-ups

- `packages/system/kubevirt-cdi/Makefile`'s `update:` target still opens with `rm -rf templates`, and `templates/` holds `cdi-uploadproxy-ingress.yaml` and `cdi-uploadproxy-tlsroute.yaml` next to the upstream `cdi-cr.yaml`, so a refresh deletes both cozystack-authored templates. The sibling `kubevirt` Makefile solved this with an idempotent re-apply plus per-marker sanity checks. Adding `test:` here at least turns that loss into a red suite instead of a silent one, but the destructive `update:` deserves its own change.
- The body names the structural fix, routing these reads through `cozy-lib` whose helpers already guard with `| default dict`, then declares the keys per chart instead. Reasonable for ten charts. The fifteen remaining ones in `apps/` and `extra/` are where the per-chart shape stops paying, and I confirmed that list matches the one the body gives.
